zkvm/src/predicate.rs

```diff
@@ -54,12 +46,33 @@ impl Predicate {
         let gens = PedersenGens::default();
         t.commit_bytes(b"prog", &prog);
         let h = t.challenge_scalar(b"h");
-        Predicate::Opaque((h * gens.B_blinding).compress())
+        Predicate::new((h * gens.B_blinding).compress())
```

Comment:
Note for later: this method as-is does not make much sense: it takes in a bytecode `[u8]` instead of `[Instruction]`, and creates an opaque witness-less predicate. We'll probably need this to take `Vec<Instruction>` and retain the witness. But we don't need to worry about this in this PR.

Reply:
would we use this method to construct the Predicate object on the verifier's side?

Reply:
ah, I guess in that instance we would just read the `CompressedRistretto` point and use that, nvm !